assets/src/edit-story/components/library/panes/pageTemplates/templateSave.js

```diff
@@ -80,38 +99,79 @@ const StyledText = styled(Text)`
   margin-bottom: 8px;
 `;
 
-const StyledButton = styled(Button)`
-  height: 32px;
-`;
+function TemplateSave({ pageSize, setShowDefaultTemplates, loadTemplates }) {
+  const {
+    actions: { addPageTemplate },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+
+  const { currentPage } = useStory(({ state: { currentPage } }) => ({
+    currentPage,
+  }));
 
-function TemplateSave({ pageSize }) {
   const { customPageTemplates } = useFeatures();
-  const handleSaveTemplate = useCallback(() => {}, []);
+  const handleSaveTemplate = useCallback(() => {
+    // Don't add empty page.
+    if (isDefaultPage(currentPage)) {
+      showSnackbar({
+        message: __(
+          'An empty page can’t be saved as a template. Add elements and try again.',
+          'web-stories'
+        ),
+        dismissable: true,
+      });
+    } else {
+      addPageTemplate({ ...currentPage, id: uuidv4(), title: null }).then(
+        () => {
+          showSnackbar({
+            message: __('Page template saved.', 'web-stories'),
+            dismissable: true,
+          });
+          loadTemplates?.();
+        }
+      );
```

Comment:
Should be have some kind of error handling here? A snackbar message saying the template failed?

Reply:
There will be but not as part of this PR. The description of the issue specifically states no error handling for now.